agent.go

```diff
@@ -468,8 +468,12 @@ func (n *network) addDriverWatches() {
 
 	c := n.getController()
 	for _, tableName := range n.driverTables {
-		ch, cancel := c.agent.networkDB.Watch(tableName, n.ID(), "")
 		c.Lock()
+		if c.agent == nil {
```

Comment:
No the problem is the `c.agent.networkDB` and `c.agent.driverCancelFuncs` access below will crash if c.agent is made nil during that time. That is the actual root cause for this crash.